### ISSUE TITLE
ipcMain.on() still run after background.js hot reload

### ISSUE THREAD

**Issue by zeroslope** [saltyshiomix/nextron#19]
ipcMain.on() still run after background hot reload

you can add `alert('a')` on this line
https://github.com/saltyshiomix/nextron/blob/6910b108cd6596bf493db1bd717322c742b4e536/examples/ipc-communication/renderer/pages/home.jsx#L24

and then reload background.js, click the button, you can see two alert; reload again, you can see three alert.

and after reload, the output won't appear on console.

**Comment by saltyshiomix** [comment]
@zeroslope 
Thank you for your reporting this issue 👍 

I fixed `nextron@3.5.20`:

```bash
# install latest nextron
$ npm install --global nextron@latest

# test it
$ nextron init my-app --template ipc-communication
$ cd my-app
$ yarn && yarn dev
```

**Comment by zeroslope** [comment]
Thank you for your contribution to open source.
The Ipc problem is due to the listener not deleted.

The most difficult is another problem that I can't output anything in the command after reload.

The first time I run `yarn dev`, I click ipc then output '1'
<img width="1440" alt="2019-02-09 3 00 36" src="https://user-images.githubusercontent.com/10218146/52517836-524f3400-2c7c-11e9-964a-70f3468215d1.png">

I change it to '2', then reload, It output nothing.

<img width="720" alt="2019-02-09 3 01 11" src="https://user-images.githubusercontent.com/10218146/52517847-9b9f8380-2c7c-11e9-9ae8-41f02bd67356.png">

I don't know why.

**Comment by saltyshiomix** [comment]
I just published `nextron@3.6.0` and fixed this issue 👍 

```bash
$ npm i -g nextron@latest

$ nextron init test-ipc -t ipc-communication
$ cd test-ipc
$ yarn && yarn dev
```

**Comment by saltyshiomix** [comment]
@zeroslope 

Thank you for your details 💯 

I got it like this:
(this is a little tricky)

1. first electron process is from `build/nextron-dev.js`
2. the electron processes after hot reloading are from `main/helpers/enable-hot-reload.js`

So this missing console output bug happens.

I changed `main/helpers/enable-hot-reload.js` to `main/helpers/exit-on-change.js` and always start the electron process from `build/nextron-dev.js`.
